sdk/python/tests/integration/e2e/test_universal_e2e.py

```diff
@@ -0,0 +1,142 @@
+import math
+from datetime import datetime, timedelta
+from typing import Optional
+
+import pandas as pd
+import pytest
+from pytz import utc
+
+from feast import FeatureStore, FeatureView
+from tests.integration.feature_repos.test_repo_configuration import TestRepoConfig as C
+from tests.integration.feature_repos.test_repo_configuration import (
+    construct_feature_store,
+)
+
+configs = [
+    C(),
+    C(
+        offline_store_creator="tests.integration.feature_repos.universal.data_sources.redshift.RedshiftDataSourceCreator",
+        online_store={"type": "dynamodb", "region": "us-west-2"},
+    ),
+    C(
+        offline_store_creator="tests.integration.feature_repos.universal.data_sources.bigquery.BigQueryDataSourceCreator",
+        online_store="datastore",
+    ),
+]
```

Comment:
I think the idea behind universal repos is that you shouldn't have to come up with configs per test file. Can we make these common parameterizations part of the interface? (in a way that new offline store doesn't require adding the config in bunch of places)

Reply:
Also for the e2e test we wanted to vary providers. Here all configs use local provider.

Reply:
I agree that the parametrization should be part of the interface. I imagine that instead of `@pytest.mark.parametrize("config", configs)`, it would look like `@pytest.mark.parametrize("feature_store", generate_feature_stores(...))`. However, it's not obvious to me what the interface for `generate_feature_stores` should look like; one challenge is that different tests want to parametrize across different components of the FeatureStore (e2e across provider, registration across registry, etc.).